openedx_events/content_authoring/data.py

```diff
@@ -54,3 +54,33 @@ class CourseCatalogData:
     schedule_data = attr.ib(type=CourseScheduleData)
     hidden = attr.ib(type=bool, default=False)
     invitation_only = attr.ib(type=bool, default=False)
+
+
+@attr.s(frozen=True)
+class XBlockBaseData:
+    """
+    Data about changed xblock.
+
+    Arguments:
+        usage_key (UsageKey): identifier of the XBlock object.
+        block_type (str): type of block.
+    """
+
+    # basic identifiers
+    usage_key = attr.ib(type=UsageKey)
+    block_type = attr.ib(type=str)
+
+
+@attr.s(frozen=True)
+class XBlockDuplicatedData(XBlockBaseData):
```

Comment:
Do we really need 2 data classes here? Can't we simply have just `XBlockData` and have the `source_usage_key` set to `None` for the "non-duplicate" events?

Reply:
@tecoholic True, just kept it separate as having `source_usage_key` for publish and delete events did not make any sense. Also for duplicate event `source_usage_key` is required which we cannot enforce if it has a default value of `None`.

Reply:
@navinkarkera I see. Thank you for the explanation.

Reply:
I'm not sure I understand the purpose of XBlockDuplicatedData.

Reply:
@mariajgrimaldi We want to react in course-discovery/taxonomy whenever an xblock is duplicated.  The idea is to store tags for a xblock in taxonomy-connector (tags are fetched from an external API by sending text representation of the xblock). Now if an xblock is duplicated, we plan to copy the tags over to the new xblock and skip sending to external API. Hence we need the source xblocks usage_key to get the tags from.

Reply:
Thanks for the explanation! Now, I wonder if we should use DuplicatedXblockData instead of XblockDuplicatedData, since it seems that the 1st indicates data from a duplicated block and the 2nd that the data is duplicated? what do you think?

Reply:
@mariajgrimaldi Thanks! the name `DuplicatedXBlockData` makes more sense. Updated!